spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/flows/ClientCredentialsTokenFlow.java

```diff
@@ -0,0 +1,238 @@
+package com.sap.cloud.security.xsuaa.token.flows;
+
+import static com.sap.cloud.security.xsuaa.token.flows.XsuaaTokenFlowsUtils.addAcceptHeader;
+import static com.sap.cloud.security.xsuaa.token.flows.XsuaaTokenFlowsUtils.addBasicAuthHeader;
+import static com.sap.cloud.security.xsuaa.token.flows.XsuaaTokenFlowsUtils.buildAuthorities;
+
+import java.net.URI;
+import java.util.Map;
+
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.util.Assert;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * A client credentials flow builder class. Applications retrieve an instance of
+ * this builder from {@link XsuaaTokenFlows} and then create the flow request
+ * using a builder pattern.
+ */
+public class ClientCredentialsTokenFlow {
+
+	private static final String ACCESS_TOKEN = "access_token";
+	private static final String GRANT_TYPE = "grant_type";
+	private static final String CLIENT_CREDENTIALS = "client_credentials";
+	private static final String AUTHORITIES = "authorities";
+
+	private RestTemplate restTemplate;
+	private XsuaaTokenFlowRequest request;
+	private VariableKeySetUriTokenDecoder tokenDecoder;
+
+	/**
+	 * Creates a new instance.
+	 * 
+	 * @param restTemplate
+	 *            - the {@link RestTemplate} used to execute the final request.
+	 * @param xsuaaBaseUri
+	 *            - the base URI of XSUAA. Based on the base URI the tokenEndpoint,
+	 *            authorize and key set URI will be derived.
+	 */
+	ClientCredentialsTokenFlow(RestTemplate restTemplate, VariableKeySetUriTokenDecoder tokenDecoder,
+			URI xsuaaBaseUri) {
+		Assert.notNull(restTemplate, "RestTemplate must not be null.");
+		Assert.notNull(xsuaaBaseUri, "XSUAA base URI must not be null.");
+		Assert.notNull(tokenDecoder, "TokenDecoder must not be null.");
+
+		this.restTemplate = restTemplate;
+		this.tokenDecoder = tokenDecoder;
+
+		URI tokenEndpoint = UriComponentsBuilder.fromUri(xsuaaBaseUri).path("/oauth/token").build().toUri();
+		URI authorizeEndpoint = UriComponentsBuilder.fromUri(xsuaaBaseUri).path("/oauth/authorize").build().toUri();
+		URI keySetEndpoint = UriComponentsBuilder.fromUri(xsuaaBaseUri).path("/token_keys").build().toUri();
+
+		this.request = new XsuaaTokenFlowRequest(tokenEndpoint, authorizeEndpoint, keySetEndpoint);
```

Comment:
call the other constructor

Reply:
How do you suggest I do that? The other constructor checks that the parameters are not null. And that is so on purpose. If I did not check the last two parameters for null values, I would not have to override the constructor in the first place. But I think it makes sense to have both constructors and also to check if the parameters are not null.